test/e2e/validation_test.go

```diff
@@ -289,13 +288,31 @@ func (tc *testContext) sshSetup() error {
 	return nil
 }
 
-// runSSHJob creates and waits for a Kubernetes job to run. The command provided will be executed on the host specified
-// by the provided IP.
-func (tc *testContext) runSSHJob(name, command, ip string) (string, error) {
+// runPowerShellSSHJob creates and waits for a Kubernetes job to run. The command provided will be executed through
+// PowerShell, on the host specified by the provided IP.
+func (tc *testContext) runPowerShellSSHJob(name, command, ip string) (string, error) {
+	keyMountDir := "/private-key"
+	sshCommand := []string{"bash", "-c",
+		fmt.Sprintf(
+			// first determine if the host has PowerShell or cmd as the default shell by running a simple PowerShell
+			// command. If it succeeds, then the host's default shell is PowerShell
+			"if ssh -o StrictHostKeyChecking=no -i %s %s@%s 'Get-Help';"+
+				"then CMD_PREFIX=\"\";CMD_SUFFIX=\"\";"+
+				// if PowerShell is not the default shell, explicitly run the command through PowerShell
+				"else CMD_PREFIX=\""+windows.RemotePowerShellCmdPrefix+" -Command \\\"\";CMD_SUFFIX=\"\\\"\";"+
```

Comment:
nit: don't think the extra space before `-Command` is needed as the `windows.RemotePowerShellCmdPrefix` const already ends in a space.